packages/edit-site/src/components/style-book/index.js

```diff
@@ -178,7 +145,31 @@ function getExamples() {
 			blocks: getBlockFromExample( blockType.name, blockType.example ),
 		} ) );
 
-	return [ headingsExample, ...otherExamples ];
+	const isHeadingBlockRegistered = !! getBlockType( 'core/heading' );
+
+	if ( ! isHeadingBlockRegistered ) {
+		return nonHeadingBlockExamples;
+	}
+
+	// Use our own example for the Heading block so that we can show multiple
+	// heading levels.
+	const headingsExample = {
+		name: 'core/heading',
+		title: __( 'Headings' ),
+		category: 'text',
+		blocks: [ 1, 2, 3, 4, 5, 6 ].map( ( level ) => {
+			return createBlock( 'core/heading', {
+				content: sprintf(
+					// translators: %d: heading level e.g: "1", "2", "3"
+					__( 'Heading %d' ),
+					level
+				),
+				level,
+			} );
+		} ),
```

Comment:
This is not directly related to this PR, but it is a code refactoring. The translator comment is the same as that of the `HeadingLevelDropdown` component:

https://github.com/WordPress/gutenberg/blob/27fbbd8e9b57495339a077ef0c8ba67a18392699/packages/block-editor/src/components/block-heading-level-dropdown/index.js#L55-L59